solver/llbsolver/vertex.go

```diff
@@ -69,21 +69,27 @@ func WithCacheSources(cms []solver.CacheManager) LoadOpt {
 
 func RuntimePlatforms(p []specs.Platform) LoadOpt {
 	var defaultPlatform *pb.Platform
+	var normalizedPlatform specs.Platform
 	pp := make([]specs.Platform, len(p))
 	for i := range p {
 		pp[i] = platforms.Normalize(p[i])
 	}
 	return func(op *pb.Op, _ *pb.OpMetadata, opt *solver.VertexOptions) error {
 		if op.Platform == nil {
 			if defaultPlatform == nil {
-				p := platforms.DefaultSpec()
-				defaultPlatform = &pb.Platform{
-					OS:           p.OS,
-					Architecture: p.Architecture,
-				}
+				normalizedPlatform = platforms.Normalize(platforms.DefaultSpec())
 			}
-			op.Platform = defaultPlatform
+		} else {
+			platform := specs.Platform{OS: op.Platform.OS, Architecture: op.Platform.Architecture, Variant: op.Platform.Variant}
+			//A typical case for normalization is variant for "arm64/v8" will be updated to "arm64" to consistent with "pp.Variant" that is passed in.
+			normalizedPlatform = platforms.Normalize(platform)
+		}
+		defaultPlatform = &pb.Platform{
```

Comment:
defaultPlatform var should only contain the default platform for current machine, not be overridden with the platform from llb.

Reply:
I agree, pls take a look at those line https://github.com/moby/buildkit/blob/master/solver/llbsolver/vertex.go#L80-L82, 

The code doesn't include the variant, so those info will be missed here for matched platforms.
https://github.com/moby/buildkit/blob/master/solver/llbsolver/vertex.go#L90

This is kind of bug per my understanding, we need to the change accommodate both the following case, 

- arm64/v8, variant is normalized to  "empty" string
- arm32/v6 arm32/v7, variants should be included to differentiate the platform information.

Reply:
The difference is that old code sets `defaultPlatform` once based on the current system. While yours sets it based on previous `op.Platform` values.

Reply:
hmm, that is a little tricky, we need the workaround here to address some edge case, I need to get the variant information from `op.platform` and then normalized it to consistent with [https://github.com/moby/buildkit/blob/master/solver/llbsolver/vertex.go#L74](url)

but, yes, I change the meaning of `defaultPlatform`, I properly need to define another var for it instead of using `defaultPlatform`.